lib/std/array_hash_map.zig

```diff
@@ -773,9 +773,9 @@ pub fn ArrayHashMapUnmanaged(
                 }
             }
 
+            try self.entries.ensureTotalCapacity(allocator, new_capacity);
             const new_bit_index = try IndexHeader.findBitIndex(new_capacity);
             const new_header = try IndexHeader.alloc(allocator, new_bit_index);
```

Comment:
I wonder if this could be a single memory allocation

Reply:
The fields for ArrayHashMap are:

```zig
        /// It is permitted to access this field directly.
        entries: DataList = .{},

        /// When entries length is less than `linear_scan_max`, this remains `null`.
        /// Once entries length grows big enough, this field is allocated. There is
        /// an IndexHeader followed by an array of Index(I) structs, where I is defined
        /// by how many total indexes there are.
        index_header: ?*IndexHeader = null,
```

Having the header be a separate allocation is an optimization for maps that are below a certain size threshold.

You are welcome to experiment with changing the memory layout here - if you can demonstrate a perf enhancement without breaking the API, that's an easy merge.